### PR TITLE
Upgrade `arrow-cpp` and `pyarrow` to `7.0.0`

### PR DESCRIPTION
**Pull request by galipremsagar**
Changes needed for: https://github.com/rapidsai/cudf/pull/10503

### CODE REVIEW

**Comment by galipremsagar** [comment]
rerun tests

File: conda/recipes/versions.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ build_stack_version:
 
 # Shared versions across meta-pkgs
 arrow_version:
-  - '=6.0.1'
+  - '=7.0.0'
```
Comment (galipremsagar) [inline conda/recipes/versions.yaml:28]:
```suggestion
  - '=7.0.0'
  
```

File: conda/recipes/versions.yaml [hunk 2]
```diff
@@ -25,7 +25,8 @@ build_stack_version:
...
+
```
Comment (galipremsagar) [inline conda/recipes/versions.yaml:29]:
```suggestion
  - '=7.0.0'
```

**Comment by galipremsagar** [comment]
rerun tests

**Comment by ajschmidt8** [comment]
The PR build job is running below for dfaf1d6. I'm not sure why the status is not being reported on this PR. Could be related to the GitHub issues that have been occurring over the last few days

- https://gpuci.gpuopenanalytics.com/job/rapidsai/job/integration/job/prb/job/integration-prb/1006/

**Comment by ajschmidt8** [comment]
Actually, on second thoughts, after checking the logs. It seems that our comments triggered CI for commit 34636814a96b4b1c7c51f5d15df8c08b71dba043 instead of the latest commit :thinking:. Not sure what's going on here. Likely still GitHub related though

**Comment by ajschmidt8** [comment]
> Actually, on second thought, after checking the logs. It seems that our comments triggered CI for commit [3463681](https://github.com/rapidsai/integration/commit/34636814a96b4b1c7c51f5d15df8c08b71dba043) instead of the latest commit thinking. Not sure what's going on here. Likely still GitHub related though

I'll work on getting that published.

**Comment by ajschmidt8** [comment]
@galipremsagar, it seems that `22.06` packages are now available: https://anaconda.org/rapidsai-nightly/xgboost/files

**Comment by galipremsagar** [comment]
> @galipremsagar, it seems that `22.06` packages are now available: https://anaconda.org/rapidsai-nightly/xgboost/files

Thanks for working on bringing them up AJ!

**Comment by ajschmidt8** [comment]
@galipremsagar. correction! still waiting on ARM packages

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
Looks like `22.06` packages aren't published for `cugraph`:

```
14:27:32 conda.exceptions.ResolvePackageNotFound: 
14:27:32   - cuml=22.06
14:27:32   - cugraph=22.06
14:27:32   - pylibcugraph=22.06
14:27:32   - libcugraph_etl=22.06
```

**Comment by galipremsagar** [comment]
rerun tests

**Comment by ajschmidt8** [comment]
admin merging. the relevant tests have passed and https://github.com/rapidsai/cudf/pull/10503 is passing CI successfully with the new version.

**Pull request merged**